Article Analyzer - Voloshkevych

Voloshkevych opened the pull request.

Voloshkevych committed:
This does not get right categories, but I am really close

Voloshkevych committed:
Ready to go

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/com/inventorsoft/homework/config/AppConfig.java:
```diff
@@ -0,0 +1,20 @@
+  @Value("${excluded.words}")
+  private String excludedWords;
+
+  @Bean
+  public Set<String> excludedWordsSet() {
+    return new HashSet<>(Arrays.asList(excludedWords.split(",")));
+  }
```
why bean of strings? I see it in every second PR... just use @Value in the class it is intented

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/com/inventorsoft/homework/service/ArticleAnalyzerService.java:
```diff
@@ -0,0 +1,45 @@
+@Service
+public class ArticleAnalyzerService {
+
+  @Autowired
+  private ArticleService articleService;
+
+  @Autowired
+  private EmailNotificationService emailService;
+
+  @Autowired
+  private SlackNotificationService slackService;
+
+  @Autowired
+  private JsonUtil jsonUtil;
```
prefer constructor injection

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/com/inventorsoft/homework/service/NotificationService.java:
```diff
@@ -0,0 +1,6 @@
+package com.inventorsoft.homework.service;
+
+import com.inventorsoft.homework.entity.User;
+public interface NotificationService {
+  void notifyUser(User user, String message);
```
add empty line here

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/com/inventorsoft/homework/utility/JsonUtil.java:
```diff
@@ -0,0 +1,43 @@
+public class JsonUtil {
+
+  public <T> List<T> readFromJson(String fileName, Class<T[]> clazz) {
+    Gson gson = new Gson();
```
make Gson as a spring bean and inject it in classes, don't create the instance every time the method is called

vitaliiStefanchakInventorsoft reviewed (commented):
nice work, left few comments to fix

Voloshkevych committed:
1. Used @Value directly where it was intended.
2. Refactored ArticleAnalyzerService to use constructor injection instead of field injection.
3. Added empty line for better readability.
4. Created config class for Gson as a spring bean and injected it in JsonUtil.

vitaliiStefanchakInventorsoft commented:
![ConsiderItDoneItsDoneGIF](https://github.com/vitaliiStefanchakInventorsoft/spring-lecture/assets/103140969/dc757d06-d1e5-4921-bcf0-a1441f2263f9)
